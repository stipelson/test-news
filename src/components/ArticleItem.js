import React from 'react';
import PropTypes from 'prop-types';
/* import Row from 'emerald-ui/lib/Row';
import Col from 'emerald-ui/lib/Col'; */

const ArticleItem = ({ title, body, image, url }) => {
  return (
    <section>
      <a
        className="list-item"
        href={url}
        target="_blank"
        rel="noopener noreferrer"
      >
        <img
          src={image}
          alt={title || 'Article image'}
          className="list-item-image"
        />
        <h3>{title}</h3>
        <p>{body}</p>
      </a>
    </section>
  );
};

ArticleItem.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string,
  image: PropTypes.string,
  url: PropTypes.string,
};

export default ArticleItem;
