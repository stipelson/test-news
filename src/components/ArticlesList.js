import React from 'react';
import PropTypes from 'prop-types';
import Row from 'emerald-ui/lib/Row';
import Col from 'emerald-ui/lib/Col';
import ArticleItem from '../components/ArticleItem';

const ArticlesList = ({ articles }) => {
  let arrayArticles = Array.isArray(articles) ? articles : [];
  return (
    <Row className="extra-p-md-6">
      {arrayArticles.map((article, index) => (
        <Col key={`article-${index}`} xs={6} className="mb-46">
          <ArticleItem
            title={article.title}
            body={article.body}
            image={article.image}
            url={article.url}
          />
        </Col>
      ))}
    </Row>
  );
};

ArticlesList.propTypes = {
  articles: PropTypes.array,
  loading: PropTypes.bool,
};

ArticlesList.defaultProps = {
  articles: [],
  loading: false,
};

export default ArticlesList;
