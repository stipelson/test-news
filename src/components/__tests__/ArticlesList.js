import React from 'react';
import { create } from 'react-test-renderer';
import sinon from 'sinon';
import Row from 'emerald-ui/lib/Row';

import ArticlesList from '../ArticlesList';
import ArticleItem from '../ArticleItem';

describe('ArticlesList: render', () => {
  it('No passed articles and render correctly', () => {
    const component = create(<ArticlesList />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const items = instance.findAllByType(ArticleItem);
    expect(items.length).toBe(0);
  });

  it('Row has the class', () => {
    const component = create(<ArticlesList />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const row = instance.findByType(Row);
    expect(row.props.className).toEqual('extra-p-md-6');
  });

  it('Passed articles and render list correctly', () => {
    const article = {
      body: '- The advancements in the field  ...',
      image:
        'https://cdn.pixabay.com/photo/2016/03/21/23/25/link-1271843_960_720.png',
      title: 'Constant Technological ...',
      url: 'https://www.google.com',
    };

    const articles = [article, article, article, article];

    const component = create(<ArticlesList articles={articles} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    // Expected list of articles
    const items = instance.findAllByType(ArticleItem);
    expect(items.length).toBe(articles.length);
  });
});

describe('ArticlesList: throws error when', () => {
  let error;

  beforeEach(() => {
    error = sinon.stub(console, 'error');
  });

  afterEach(() => {
    error.restore();
  });

  it('Not-array-articles is passed to articles list', () => {
    const articles = 'This is a string';

    create(<ArticlesList articles={articles} />);

    expect(error.called).toEqual(true);
  });
});
