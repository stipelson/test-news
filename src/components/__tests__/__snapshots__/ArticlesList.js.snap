// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList: render No passed articles and render correctly 1`] = `
<div
  className="eui-row extra-p-md-6"
  data-testid="article-list"
/>
`;

exports[`ArticlesList: render Passed articles and render list correctly 1`] = `
<div
  className="eui-row extra-p-md-6"
  data-testid="article-list"
>
  <div
    className="eui-col mb-46 eui-col-xs-6"
    data-testid="article-item"
  >
    <section>
      <a
        className="list-item"
        href="https://www.google.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Constant Technological ..."
          className="list-item-image"
          src="https://cdn.pixabay.com/photo/2016/03/21/23/25/link-1271843_960_720.png"
        />
        <h3>
          Constant Technological ...
        </h3>
        <p>
          - The advancements in the field  ...
        </p>
      </a>
    </section>
  </div>
  <div
    className="eui-col mb-46 eui-col-xs-6"
    data-testid="article-item"
  >
    <section>
      <a
        className="list-item"
        href="https://www.google.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Constant Technological ..."
          className="list-item-image"
          src="https://cdn.pixabay.com/photo/2016/03/21/23/25/link-1271843_960_720.png"
        />
        <h3>
          Constant Technological ...
        </h3>
        <p>
          - The advancements in the field  ...
        </p>
      </a>
    </section>
  </div>
  <div
    className="eui-col mb-46 eui-col-xs-6"
    data-testid="article-item"
  >
    <section>
      <a
        className="list-item"
        href="https://www.google.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Constant Technological ..."
          className="list-item-image"
          src="https://cdn.pixabay.com/photo/2016/03/21/23/25/link-1271843_960_720.png"
        />
        <h3>
          Constant Technological ...
        </h3>
        <p>
          - The advancements in the field  ...
        </p>
      </a>
    </section>
  </div>
  <div
    className="eui-col mb-46 eui-col-xs-6"
    data-testid="article-item"
  >
    <section>
      <a
        className="list-item"
        href="https://www.google.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Constant Technological ..."
          className="list-item-image"
          src="https://cdn.pixabay.com/photo/2016/03/21/23/25/link-1271843_960_720.png"
        />
        <h3>
          Constant Technological ...
        </h3>
        <p>
          - The advancements in the field  ...
        </p>
      </a>
    </section>
  </div>
</div>
`;

exports[`ArticlesList: render Row has the class 1`] = `
<div
  className="eui-row extra-p-md-6"
  data-testid="article-list"
/>
`;
